lyse/__init__.py

```diff
@@ -422,6 +423,11 @@ def __init__(self, h5_path, run_paths, no_write=False):
             run_paths = run_paths['filepath']
         self.h5_path = h5_path
         self.no_write = no_write
+        try:
+            h5py.File(h5_path, 'r')
```

Comment:
I think that this may leave an open reference to the h5 file. It would be better to use a context manager like this: 

``` 
with h5py.File(h5_path, 'r') as f:
    pass
```
Which ensures that the file is closed appropriately.